verify/verify.go

```diff
@@ -866,20 +866,52 @@ func verifyHash256(quote *pb.QuoteV4) error {
 	return nil
 }
 
-func isSvnHigherOrEqual(svn []byte, components []pcs.TcbComponent) bool {
-	for i := range svn {
-		if svn[i] < components[i].Svn {
+func isCPUSvnHigherOrEqual(pckCertCPUSvnComponents []byte, sgxTcbcomponents []pcs.TcbComponent) bool {
+	for i := range pckCertCPUSvnComponents {
+		if pckCertCPUSvnComponents[i] < sgxTcbcomponents[i].Svn {
 			return false
 		}
 	}
 	return true
 }
 
+func isTdxTcbSvnHigherOrEqual(teeTcbSvn []byte, tdxTcbcomponents []pcs.TcbComponent) bool {
+	start := 0
+	if teeTcbSvn[0] > 0 {
+		start = 2
+	}
+	for i := start; i < len(teeTcbSvn); i++ {
```

Comment:
If the len(teeTcbSvn) is checked as per previous comments then you don't have to worry about returning true when start = 2 and len(teeTcbSvn) = 1